qp_woltka/woltka.py

```diff
@@ -338,3 +346,191 @@ def _coverage_copy(dest):
     else:
 
         return True, ainfo, ""
+
+
+def woltka_syndna_to_array(files, output, database_bowtie2, prep, url, name):
+    """Creates files for submission of per sample bowtie2 and woltka_syndna
+    """
+    environment = environ["ENVIRONMENT"]
+    db_folder = dirname(database_bowtie2)
+    db_name = basename(database_bowtie2)
+
+    lookup = prep.set_index('run_prefix')['sample_name'].to_dict()
+    n_files = 1
+    for i, (k, (f, r)) in enumerate(files.items()):
+        if i >= n_files*TASKS_IN_SCRIPT:
+            n_files += 1
+
+        sname = search_by_filename(basename(f['filepath']), lookup)
+        line = f'fwd_{sname} {f["filepath"]}\n'
+        if r is not None:
+            line += f'rev_{sname} {r["filepath"]}\n'
+
+        with open(join(output, f'sample_details_{n_files}.txt'), 'a+') as fh:
+            fh.write(line)
+
+    # Bowtie2 command structure based on
+    # https://github.com/BenLangmead/bowtie2/issues/311
+    bowtie2 = f'bowtie2 -p {PPN} -x {database_bowtie2} ' + \
+              '-q ${f} -S $PWD/sams/${sn}.sam ' +\
+              '--seed 42 --very-sensitive -k 16 --np 1 --mp "1,1" ' + \
+              '--rdg "0,1" --rfg "0,1" --score-min "L,0,-0.05" ' + \
+              '--no-head --no-unal --un-gz $PWD/reads/${fn}'
+
+    # all the setup pieces
+    lines = ['#!/bin/bash',
+             '#SBATCH -p qiita',
+             '#SBATCH --mail-user "qiita.help@gmail.com"',
+             f'#SBATCH --job-name {name}',
+             '#SBATCH -N 1',
+             f'#SBATCH -n {PPN}',
+             f'#SBATCH --time {SYNDNA_WALLTIME}',
+             f'#SBATCH --mem {SYNDNA_MEMORY}',
+             f'#SBATCH --output {output}/{name}_%a.log',
+             f'#SBATCH --error {output}/{name}_%a.err',
+             f'#SBATCH --array 1-{n_files}%{MAX_RUNNING}',
+             f'cd {output}',
+             'mkdir -p reads sams',
+             f'{environment}',
+             'date',  # start time
+             'hostname',  # executing system
+             'echo ${SLURM_JOBID} ${SLURM_ARRAY_TASK_ID}',
+             f'dbbase={db_folder}',
+             f'dbname={db_name}',
+             f'output={output}',
+             'while read -r sn f;',
+             '  do',
+             '    fn=`basename $f`; ',
+             f'    {bowtie2}',
+             '  done < sample_details_${SLURM_ARRAY_TASK_ID}.txt',
+             'date']
+
+    # write out the script
+    main_fp = join(output, f'{name}.slurm')
+    with open(main_fp, 'w') as job:
+        job.write('\n'.join(lines))
+
+    memory = ceil(n_files * .8)
+    time = ceil(n_files * 15)
+    # creating finish job
```

Comment:
jinja2 might also be a good way of implementing the job script creation, but I don't mind either way.

Reply:
my opinion is: if it's working and there is no actual need to change, why change it?!

Reply:
I would agree with that.